### PR TITLE
Refs [TASK][34629] Handle event Current Weather Screen Logic and UI

### PR DESCRIPTION
**Pull request by I-am-Lien**
## Related Tickets
- [#Task 34629: Content](https://edu-redmine.sun-asterisk.vn/issues/34629)
## WHAT
- Handle event Current Weather Screen Logic and UI
## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/li/weatherapp/data/source/remote/api/APIQueries.kt [hunk 1]
```diff
@@ -17,4 +17,22 @@ object APIQueries {
...
+
+    fun queryAQIForecast(lat: String, lon: String) =
+        Uri.Builder().scheme(APIConstants.SCHEME_HTTPS)
+            .authority(APIConstants.OPENWEATHER_AUTHORITY_API)
+            .appendPath(APIConstants.OPENWEATHER_CONTENT_AIR_POLLUTION)
+            .appendQueryParameter(APIConstants.OPENWEATHER_LAT, lat)
+            .appendQueryParameter(APIConstants.OPENWEATHER_LON, lon)
+            .appendQueryParameter(APIConstants.OPENWEATHER_API_KEY, BuildConfig.OPENWEATHER_API_KEY)
+            .toString()
+    fun queryCurrentWeatherForecast(lat: String, lon: String) =
```
Comment (huytq-2422) [inline app/src/main/java/com/li/weatherapp/data/source/remote/api/APIQueries.kt]: thêm 1 dòng trống cho thoải mái

File: app/src/main/java/com/li/weatherapp/utils/AirPollutionUtils.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    fun getAirPollutionTitle(aqiDegree: Int, context: Context) = when (aqiDegree) {
+        in 0..19 -> context.resources.getString(R.string.title_wonderful)
+        in 20..49 -> context.resources.getString(R.string.title_good)
+        in 50..99 -> context.resources.getString(R.string.title_bad)
+        in 100..149 -> context.resources.getString(R.string.title_harmful)
+        in 150..249 -> context.resources.getString(R.string.title_very_harmful)
+        else -> context.resources.getString(R.string.title_dangerous)
+    }
+
+    fun getAirPollutionDescription(aqiDegree: Int, context: Context) = when (aqiDegree) {
```
Comment (huytq-2422) [inline app/src/main/java/com/li/weatherapp/utils/AirPollutionUtils.kt]: sai conventions 
https://github.com/framgia/coding-standards/blob/master/eng/android/codingstyleguide.md#29-parameter-ordering-in-methods

File: app/src/main/java/com/li/weatherapp/utils/SharePreferenceExtention.kt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package com.li.weatherapp.utils
+
+import android.content.SharedPreferences
+
+fun SharedPreferences.Editor.putDouble(key: String, double: Double): SharedPreferences.Editor =
+    putLong(key, java.lang.Double.doubleToRawLongBits(double))
+
+fun SharedPreferences.getDouble(key: String, default: Double) =
+    java.lang.Double.longBitsToDouble(getLong(key, java.lang.Double.doubleToRawLongBits(default)))
```
Comment (huytq-2422) [inline app/src/main/java/com/li/weatherapp/utils/SharePreferenceExtention.kt]: kotlin Double không có hàm longBitsToDouble hay sao thế ?

File: app/src/main/java/com/li/weatherapp/utils/SharePreferenceHelper.kt [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    private var sharedPreferences =
+        context.getSharedPreferences(Constants.PREF_NAME, Context.MODE_PRIVATE)
+
+    fun <T> put(key: String, data: T) {
+        val editor = sharedPreferences.edit()
+        when (data) {
+            is String -> editor.putString(key, data)
+            is Double -> editor.putDouble(key, data)
+        }
+        editor.apply()
```
Comment (huytq-2422) [inline app/src/main/java/com/li/weatherapp/utils/SharePreferenceHelper.kt:16]: Tại sao lại dùng apply thay vì commit thế ?

File: gradle.properties [hunk 5]
```diff
@@ -19,4 +19,4 @@ android.useAndroidX=true
 android.enableJetifier=true
 # Kotlin code style for this project: "official" or "obsolete":
 kotlin.code.style=official
-OPENWEATHER_API_KEY=API_KEY
+OPENWEATHER_API_KEY=16f2019b3a524da8debc32c291c5e838
```
Comment (huytq-2422) [inline gradle.properties]: key

File: app/src/main/java/com/li/weatherapp/utils/SharePreferenceHelper.kt (hunk 4 above)
Comment (I-am-Lien) [reply app/src/main/java/com/li/weatherapp/utils/SharePreferenceHelper.kt:16]: Sử dụng apply vì tôi không cần lấy boolean trả về và k block main thread

**Review by huytq-2422 — APPROVED** [review]

**Pull request merged**